File: backend/api/Models/Deck.cs
using api.DTO;
using System.Text.Json.Serialization;

namespace api.Models;

public class Deck
{
    public int Id { get; set; }
    public string Name { get; set; }
    public ICollection<Card> Cards { get; set; } = new List<Card>();
    public int UserId { get; set; }
    [JsonIgnore]
    public User User { get; set; }

    public Deck() {}

    public Deck(DeckCreateRequest deckCreateRequest)
    {
        Name = deckCreateRequest.Name;
        Cards = deckCreateRequest.Cards
            .Select(card => new Card(card))
            .ToList();
    }
}

File: mobile/lib/models/deck.dart
import 'package:mobile/models/card.dart';

class Deck {
  int id;
  String name;
  List<Card> cards;

  Deck({
    required this.id,
    required this.name,
    required this.cards,
  });

  factory Deck.fromJson(Map<String, dynamic> json) => Deck(
        id: json["id"],
        name: json["name"],
        cards: List<Card>.from(json["cards"].map((x) => Card.fromJson(x))),
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "name": name,
        "cards": List<dynamic>.from(cards.map((x) => x.toJson())),
      };
}
